### ISSUE TITLE
Flickity Touch events in mobile Chrome break

### ISSUE THREAD

**Issue by exophunk** [metafizzy/flickity#540]
Hi everyone,

I figured out that flickity stopped swiping on mobile chrome (android) and also in the chrome dev tools mobile simulator. The arrow buttons and flickity in general still work, but no swiping is possible. Either nothing happens or weird things. Desktop Chrome, iOS and everything else is fine.

Multiple older projects (that worked with the exact same settings ~weeks ago, no changes!) stopped working! 

This happens only when I use flickity via the npm package `npm install flickity` and `import Flickity from 'flickity'`.

It doesn't happen when I directly include the flickity sources via <script> tag. It happens in multiple projects with all different types of flickity usages. It has nothing todo with CSS or my configuration of flickity, I tried everything back down to a plain vanilla flickity slider. The problem is with the build process somehow:

I use laravel-mix to build my sources which includes:

– Webpack
– Bubel Transpiler

Somehow now suddently flickity totally breaks. Anyone has an idea how the build process could make this break?

Thanks,

Robert

**Comment by exophunk** [comment]
maybe this is also related: https://www.chromestatus.com/features/5093566007214080 

(try the flickity webpage in dev simulator with touch, the warning pops up all the time. It doesn't pop anymore in my case where it is breaking)

**Comment by desandro** [comment]
Thanks for reporting this bug. I'll have to take a look with the tools you mentioned.

**Comment by jlquaccia** [comment]
hi all,  was wondering about the status of this bug.  I've come across the same issue, Flickity works totally fine on desktop in Chrome but on Chrome's mobile browser touch for swiping left/right on the carousel stopped working.  Also, navigating slides via dots on chrome mobile stopped working too (both, used to work before).  I've tried a bunch of work arounds but can't get anything to work.  Anyone know of a solution to get around this?  I'm using angular-flickity https://github.com/benjamincharity/angular-flickity

**Comment by zNiik** [comment]
Hi, I'm using this plugin for my angular 2 app. works properly with desktop but in mobile unexpected things happens like the slider breaks when after the first swipe. 

Attached is the screenshot error that I'm getting after the first swipe on mobile. Currently I'm debugging it in iOS safari using the web inspector. I also tried in google chrome mobile and get the same behavior. It only happens on mobile

![screen shot 2017-04-26 at 5 59 39 pm](https://cloud.githubusercontent.com/assets/5515259/25468329/43b2cac8-2b26-11e7-8f38-8dfb8e99ab10.png)

**Comment by levkus** [comment]
Hi, everyone!
I have the same issue.
I import Flickity from npm package and swiping doesn't work on chrome mobile (or android webview), but it works on ios safari.
But it does work in my older projects that i made using the same build system.
I tried both 2.0.5 (since older projects use it) and 2.0.7 and had no luck.
Was anyone able to solve this ?

**Comment by levkus** [comment]
In case someone needs it working right now:
I managed to get it working by not only downgrading Flickity to 2.0.5,
but also downgrading unidragger and unipointer (just in case) to 2.1.0

**Comment by simeydotme** [comment]
I also took the same action as @levkus and experience blissful flickity'ng again :)

**Comment by AlchemyUnited** [comment]
1) I've (just) noticed this as well. It no longer works on my touch Android device - which is where I first noticed this. It also doesn't work on my touch screen laptop (Win 10, Chrome Version 58.0.3029.110 ). FWIW, I'm using Flickity via CDN.

2) That said, does anyone know which version the CDN uses? More importantly how to specify version via CDN? 

3) @levkus - First, thanks! Second, help me out (please).  What is unidragger and unipointer? TIA

4)

**Comment by levkus** [comment]
@AlchemyUnited, these the libraries, that Flickity depend on.

**Comment by AlchemyUnited** [comment]
thanks @levkus 

FYI - I reverted to 2.0.5 and everything seems fine. So far. Thanks again.

**Comment by davidhellmann** [comment]
Same Probleme here. Try now 2.0.5

**Comment by davidhellmann** [comment]
@levkus hm unidragger is in the package.json but where is the unipointer ?

**Comment by desandro** [comment]
Thanks for reporting this bug. I'll be investigating and resolving this in the next week.

For now you can downgrade to [v2.0.5](https://github.com/metafizzy/flickity/tree/v2.0.5)

``` html

<link rel="stylesheet" href="https://unpkg.com/flickity@2.0.5/dist/flickity.min.css">
<!-- or -->
<link rel="stylesheet" href="https://unpkg.com/flickity@2.0.5/dist/flickity.css">
```

``` html
<script src="https://unpkg.com/flickity@2.0.5/dist/flickity.pkgd.min.js"></script>
<!-- or -->
<script src="https://unpkg.com/flickity@2.0.5/dist/flickity.pkgd.js"></script>
```

**Comment by nicolasporion** [comment]
Hi, everyone!
I had the same issue.
I've downgraded to v2.0.5 and now it works!
Thanks to @desandro

**Comment by davidhellmann** [comment]
hm, here is not working with an downgrade to 2.0.5 but I use it as NPM Package. Maybe that's the problem?

**Comment by romain-perspective** [comment]
Same as David, I even tried downgrading to 2.0.4, downgrading unidragger and unipointer... but no success :(

**Comment by jmharveys** [comment]
@desandro any updates about this issue?

**Comment by desandro** [comment]
I've released [v2.0.8](https://github.com/metafizzy/flickity/releases/tag/v2.0.8) with a fix for this issue.

Looks like the issue was the regression in v2.0.6, that `touch-action: none` is necessary when using pointer events. Pointer Events were added in Chrome 55.

Please upgrade and chime back in if you continue to have trouble. Thank you for your feedback and patience for this issue. 🌈🐻

**Comment by romain-perspective** [comment]
Thanks a lot for the new release!!
It fixed it for me. I noticed one issue though, on Chrome mobile, I can not scroll vertically over a slider, it gets blocked. Even though I have 'dragThreshold' on. Does that make sense?

Cheers!

**Comment by desandro** [comment]
@romain-perspective Thanks for following up. I'll be fixing this in the next release.

**Comment by AlchemyUnited** [comment]
@desandro  - Great thanks!

p.s. Is there any way to control which version we use when using the CDN. Best I can tell it's "latest" but there's no way to know / specify. Moi? I'd rather choose my weapons than have them picked for me :)

**Comment by desandro** [comment]
@AlchemyUnited Yes! You can specify the version within the URL.

``` html
<link rel="stylesheet" href="https://unpkg.com/flickity@2/dist/flickity.css">
<link rel="stylesheet" href="https://unpkg.com/flickity@2/dist/flickity.css">
```

These URLs use `2`, which will deliver the latest 2.x.x version. You can specify a specific version:

``` html
<link rel="stylesheet" href="https://unpkg.com/flickity@2.0.5/dist/flickity.css">
<link rel="stylesheet" href="https://unpkg.com/flickity@2.0.5/dist/flickity.css">
```

These URLs use `2.0.5`.

**Comment by AlchemyUnited** [comment]
@desandro  - Awesome. Thanks. 

FYI - I don't recall seeing that in the docs. Maybe I missed it?

**Comment by HugoGiraudel** [comment]
We (at N26) confirm this bug happens. We’ll try downgrading to 2.0.5 to see if it fixes it.

**Comment by davidhellmann** [comment]
Bug is still there :(
You can check it here: https://fredmansky.at/p/about
No swiping on Mobile (Android Chrome, Mac Chrome Dev Tools with iPhone activated) 

:(

**Comment by HugoGiraudel** [comment]
2.0.8 did not appear to fix it as far as I can tell. :(

**Comment by desandro** [comment]
@davidhellmann Can you update to 2.0.8? Looks like you're on a previous version.

@HugoGiraudel Have you tried v2.0.8, just released? You'll have to update Unidragger to 2.2.2

**Comment by HugoGiraudel** [comment]
@desandro It appears to have fixed it. Thank you!

**Comment by felipearosemena** [comment]
@desandro I'm still not able to scroll vertically when swiping over the slides. I've upgraded to 2.0.8 & Unidragger to 2.2.2. I still see `touch-action: none` being applied on mobile.

Managed to hack a fix by doing:

 ```
.flickity-viewport {
  touch-action: initial !important;
}
```

**Comment by desandro** [comment]
>  I'm still not able to scroll vertically when swiping over the slides.

@felipearosemena Yup, this is a regression bug. I'll be adding a fix in the next release. This Hack CSS may fix it in the mean time:

``` css
.flickity-viewport {
  touch-action: pan-y !important;
}
```

**Comment by rs3d** [comment]
I'm also having issues since updating from 2.0.5 to 2.0.8. 
On desktop native image drag is often happening. I've to use pointer-events: none on images to have wanted behavior. For now i downgraded back to the older version

**Comment by desandro** [comment]
v2.0.9 has been released with a fix for the `touch-action: pan-y` issue. I recommend upgrading to this version and checking if this issue persists.

Thank you for your feedback and interest in Flickity!

**Comment by desandro** [comment]
I'm closing this issue as resolved. Please open a new issue if you continue to run into a similar problem. 🌈🐻

**Comment by vignesh-coderapper** [comment]
> I've released [v2.0.8](https://github.com/metafizzy/flickity/releases/tag/v2.0.8) with a fix for this issue.
> 
> Looks like the issue was the regression in v2.0.6, that `touch-action: none` is necessary when using pointer events. Pointer Events were added in Chrome 55.
> 
> Please upgrade and chime back in if you continue to have trouble. Thank you for your feedback and patience for this issue. 🌈🐻

Its worked for me when i have added touch-event: none to carousel-cells
